pkg/api/v20170701/validate.go

```diff
@@ -157,9 +161,18 @@ func (a *Properties) Validate() error {
 	if e := validateUniqueProfileNames(a.AgentPoolProfiles); e != nil {
 		return e
 	}
+
 	if a.OrchestratorProfile.OrchestratorType == Kubernetes {
-		if a.ServicePrincipalProfile == nil {
```

Comment:
I believe when using auto generating client secret, we are still populating the secret in the template instead of populating in KV.
@colemickens am I correct?